### PR TITLE
[CDAP-13371] Handling Spark1 environments for MMDS and Reports in UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
- MMDS and Reports app require spark2 to run successfully.
- This PR handles UX in Spark1 environments by gracefully error'ing out.

JIRA: https://issues.cask.co/browse/CDAP-13371
Build: https://builds.cask.co/browse/CDAP-URUT43

### CODE REVIEW

File: cdap-ui/app/cdap/components/Reports/ReportsServiceControl/index.js [hunk 1]
```diff
@@ -60,6 +71,16 @@ export default class ReportsServiceControl extends Component {
   };
 
   renderEnableBtn = () => {
+    if (this.state.disabled) {
+      return (
+        <div className="action-container service-disabled">
+          <IconSVG name="icon-exclamation-triangle" className="text-danger" />
+          <div className="text-danger">
+            Resports require Spark v2. Please upgrade to Spark v2 to enable reports
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Reports/ReportsServiceControl/index.js]: typo for reports. Perhaps i18n this?

File: cdap-ui/app/cdap/components/Experiments/ExperimentsServiceControl/index.js [hunk 2]
```diff
@@ -32,8 +32,18 @@ export default class ExperimentsServiceControl extends Component {
     onServiceStart: PropTypes.func
   };
 
+  componentDidMount() {
+    isSpark2Available()
+      .subscribe(
+        isAvailable => this.setState({
+          disabled: !isAvailable
+        })
+      );
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Experiments/ExperimentsServiceControl/index.js]: can there be a case where the `versions` API takes a while to return, and we are already showing the landing page?

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 3]
```diff
@@ -2107,6 +2109,8 @@ features:
         b6: Use-case based customization of report contents (columns).
       description: "Reports help users generate historical analysis of programs and pipelines running in their data infrastructure. The primary goal of these reports is to help decision makers and administrators make informed business decisions about resource utilization, capacity planning, and onboarding new processes."
       enable: Enable Reports
+      environmentCheckMessage: Checking for environment before enabling Analytics
```
Comment (elfenheart) [inline cdap-ui/app/cdap/text/text-en.yaml]: ...before enabling Reports

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Review by bdmogal — APPROVED** [review]

**Pull request merged**